lib/datastruct/kvhash.c

```diff
@@ -20,9 +21,21 @@ rehash(struct kvhash * H)
 	uint32_t * new_hashes;
 	size_t i, pos;
 
+	/* Sanity check. */
+	assert(H->nslots > 0);
+
+	/* Check for overflow. */
+	if (H->nslots >= SIZE_MAX / 2) {
+		warn0("Overflow in kvhash rehash");
+		goto err0;
+	}
+
 	/* Double the table size. */
 	new_nslots = H->nslots * 2;
 
+	/* Sanity check; clang can't figure this out by itself. */
+	assert(new_nslots > 0);
```

Comment:
am I going crazy here?  If `H->nslots > 0` and `H->nslots < SIZE_MAX / 2`, then `new_nslots` should be `> 0`, right?  Both variables are `size_t`.

Reply:
Looks right to me.